packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +97,304 @@ describe("Pooled Transaction Fees Test", function () {
 			RUNTIME_ERRORS.TRANSACTION.FEES_NOT_ENOUGH
 		);
 	});
+
+	it("Should pass if two out of three doesn't pay the txn fee in a group txn of size three", function () {
+		const amount = 4000;
+		const fee = 3000;
+		// group with fee distribution. Pooled transaction fee
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: bob.account,
+				toAccountAddr: john.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+		];
+
+		assert.doesNotThrow(() => runtime.executeTx(groupTx));
+
+		syncAccounts();
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(alice.balance(), BigInt(initialBalance));
+		assert.equal(bob.balance(), BigInt(initialBalance));
+	});
+
+	it("Should unfunded accounts be able to issue transactions and opt-in", function () {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elonUnfunded.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			},
+			{
+				type: types.TransactionType.OptInASA,
+				sign: types.SignType.SecretKey,
+				fromAccount: elonUnfunded.account, // unfunded account
+				assetID: assetId,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+		];
+
```

Comment:
assert(Elon.balance, 0). We can also do this in beforeEach to save some space

Reply:
Moved to beforeEach